### PR TITLE
Fix the library to work on non-US locale.

### LINKED ISSUE #6: vcsstate assumes LANG=en_US.UTF8
I found this problem while trying out gostatus. This breaks also your project Go-Package-Store.
## Repro

```
$ go get -u github.com/shurcooL/gostatus
$ gostatus github.com/shurcooL/gostatus
```

Running on `git version 2.9.3`, `go version go1.7.1 linux/amd64` on Ubuntu 14.04.
## Expected

States everything is up to date.
## Actual

```
$ gostatus github.com/shurcooL/gostatus
2016/09/26 10:59:38 github.com/shurcooL/gostatus: no HEAD branch
  ?  github.com/shurcooL/gostatus/...
        ? Unreachable remote (check your connection)
$ cd $GOPATH/src/github.com/shurcooL/gostatus
$ git remote -v
origin  https://github.com/shurcooL/gostatus (fetch)
origin  https://github.com/shurcooL/gostatus (push)
$ git remote show origin
* distante origin
  URL de rapatriement : https://github.com/shurcooL/gostatus
  URL push : https://github.com/shurcooL/gostatus
  Branche HEAD : master
  Branches distantes :
    experiment/offline_and_diff_flags suivi
    master                            suivi
  Branche locale configurée pour 'git pull' :
    master fusionne avec la distante master
  Référence locale configurée pour 'git push' :
    master pousse vers master (à jour)
$ echo $LANG
fr_CA.UTF-8
```
## Problem

`git.go` assumes `LANG=en_US.UTF8` and tries to parse output that is not in porcelain mode. The fix would be to force environment variable `LANG=en_US.UTF8` at line 139.

### PR DESCRIPTION
**Pull request by maruel**
Merge code duplicated 4 times.

Fixes #6.

### CODE REVIEW

File: util.go [hunk 1]
```diff
@@ -13,3 +16,35 @@ func dividedOutput(cmd *exec.Cmd) (stdout []byte, stderr []byte, err error) {
...
+	c := exec.Command(name, cmd...)
+	if len(dir) != 0 {
+		c.Dir = dir
+	}
+	dstEnv := osutil.Environ(os.Environ())
+	for k, v := range env {
+		dstEnv.Set(k, v)
+	}
+	dstEnv.Set("LANG", "C.UTF8")
+	dstEnv.Set("LANGUAGE", "C.UTF8")
```
Comment (dmitshur) [inline util.go:30]: Do both `LANG` and `LANGUAGE` need to be set to fix this issue? If we don't set one of them, will the issue persist?

Why is the value `C.UTF8`, expected it to be `en_US.UTF8` from #6. What does "C" stand for?

**Review by dmitshur — COMMENTED** [review]
Thanks! Some inline questions.

**Comment by maruel** [comment]
http://www.gnu.org/software/gettext/manual/html_node/The-LANGUAGE-variable.html and http://www.gnu.org/software/gettext/manual/html_node/Locale-Names.html are relevant.

Technically when using C LANGUAGE may not be needed but that's not 100% sure. I did recall having to do it but I forget on which platform. Keep in mind you may get interesting issues when using cygwin on Windows, msys bash on Windows, etc so stomping them all is safer. That said it's been too long since I've done this to recall the exact requirements.

That said it's likely safer to use en_US.UTF-8, which is what I had used in my own project. I wonder how this behaves on system like a Raspberry Pi with en_GB installed but running on a non-English locale. I think the fallback is likely natural but I don't know for sure.

So long story short, I can use en_US instead of C, but I think C is a safer fallback. I may be wrong.

**Comment by dmitshur** [comment]
Thanks for answering @maruel.

I'll need to think more about this and do some research before I can review this fully.

I prefer to add/change as minimum as possible to fix a concrete, reproducible bug today. If there are additional bugs in the future, they can be reported and fixed in a similar way.

The alternative is adding code that is poorly understood, possibly has no effect, and really hard to change/debug in the future.

I'll try to get to this around this weekend or so. Thanks again for the PR!

**Comment by dmitshur** [comment]
Looks like I won't be able to get to this until sometime next week. I am giving a talk at dotGo this coming Monday and have to take care of that first.

If this is time-sensitive or blocking for you, let me know. Otherwise I'll resume here next week.

Some initial thoughts. I'd like to split this change into two parts, the refactor (no behavior change) and the actual fix (only what's needed to fix #6). That way it's easier to review/comment on those separate parts.

**Comment by dmitshur** [comment]
Sorry about the delay, here's an update.

I'm able to reproduce this by using a VM, booting a Ubuntu 16.10 live image and picking a non-English language as my choice of language.

I will not be able to review this PR as is because it's a single commit that performs a refactor to deduplicate some code (to make applying the fix easier) and then applies the fix. To make it viable to review, it should be split into at least 2 commits (zero behavior change refactor, and bug fix).

However, I've realized I want to think more carefully about the refactor step and how I want to layout the code. This will be easier for me to figure out myself, rather than incurring the cost of communication latency over GitHub PR comments.

Thank you for making this PR, I appreciate it, and sorry I won't be able to accept it as is. I hope that's okay. Thanks again!
